File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s452855095_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	n, _ := strconv.Atoi(scanner.Text())
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		scanner.Scan()
		num, _ := strconv.Atoi(scanner.Text())
		arr[i] = num
	}
	cnt := 0
	flag := true
	for {
		for i := 0; i < n; i++ {
			if arr[i]%2 != 0 {
				flag = false
				break
			}
			arr[i] /= 2
		}
		if flag {
			cnt++
		} else {
			break
		}
	}
	fmt.Println(cnt)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/multiline/comment/direct/Java/C/s452855095_commented.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int arr[n];
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    int cnt = 0;
    int flag = 1;
    while (1) {
        for (int i = 0; i < n; i++) {
            if (arr[i] % 2 != 0) {
                flag = 0;
                break;
            } else {
                arr[i] = arr[i] / 2;
            }
        }
        if (flag == 1) {
            cnt++;
        } else {
            break;
        }
    }
    printf("%d\n", cnt);
} // End of Code
